### PR TITLE
[3006.x] Clarifying the documentation for extension_modules configuration option

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Clarifying the documentation for extension_modules configuration option

### What issues does this PR fix or reference?
Fixes: #63929 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: doc/ref/configuration/master.rst [hunk 1]
```diff
@@ -206,10 +206,13 @@ following the Filesystem Hierarchy Standard (FHS) might set it to
     moved into the master cachedir (on most platforms,
     ``/var/cache/salt/master/extmods``).
 
-Directory for custom modules. This directory can contain subdirectories for
-each of Salt's module types such as ``runners``, ``output``, ``wheel``,
-``modules``, ``states``, ``returners``, ``engines``, ``utils``, etc.
-This path is appended to :conf_master:`root_dir`.
+Directory wehre custom modules are synced to. This directory can contain
```
Comment (whytewolf) [inline doc/ref/configuration/master.rst:209]: typo in where

**Review by whytewolf — DISMISSED** [review]

File: conf/master [hunk 2]
```diff
@@ -60,15 +60,19 @@
 #
 #cachedir: /var/cache/salt/master
 
-# Directory for custom modules. This directory can contain subdirectories for
-# each of Salt's module types such as "runners", "output", "wheel", "modules",
-# "states", "returners", "engines", "utils", etc.
+# Directory where custom modules synced to This directory can contain
```
Comment (s0undt3ch) [inline conf/master:63]:
```suggestion
# Directory where custom modules sync to. This directory can contain
```
?

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: conf/master [hunk 3]
```diff
@@ -60,15 +60,19 @@
...
+#
+# Note, any directories or files not found in the `module_dirs`
+# location will be removed from the extension_modules path.
+
 #extension_modules: /var/cache/salt/master/extmods
 
 # Directory for custom modules. This directory can contain subdirectories for
 # each of Salt's module types such as "runners", "output", "wheel", "modules",
 # "states", "returners", "engines", "utils", etc.
-# Like 'extension_modules' but can take an array of paths
```
Comment (onmeac) [inline conf/master:71]: May change this to `must be array of pahts` as I got this message during salt-master startup: `Config option 'module_dirs' with value /srv/salt/extmods has an invalid type of str, a list is required for this option`

**Comment by onmeac** [comment]
I am not sure this fixes #63929 

Even when using `module_dirs` a salt-syndic process reports it cannot find the `get_load` function.

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**